Run search term standardise function on GA4 search event

AshGDS opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

## What

Run `standardiseSearchTerm` on the GA4 search event, so that it looks the same as all the other search terms that come through e.g. in the ecommerce and pageview events.

## Why

https://trello.com/c/J5XcLlvh/743-collect-undefined-instead-of-when-removing-search-terms-on-finders

andysellick reviewed (approved):
Nice. Does this need documenting anywhere, or did you already do enough in the gem?

AshGDS commented:
Thanks @andysellick - I have added a couple commits to this branch as I realised I can add another test, as well as updating the documentation.

I have created a PR for `govuk-developer-docs`: https://github.com/alphagov/govuk-developer-docs/pull/4335

Would you be able to review the extra commits I've added? 

Thanks :+1:
